### PR TITLE
Refactor Controller Inheritance

### PR DESCRIPTION
**Pull request by Maaiins**

### CODE REVIEW

**Comment by Maaiins** [comment]
I will do the other files the next few days. Today I got limited time

**Comment by Maaiins** [comment]
You can push your changes to this branch also. So we can review them together.

File: modules/base/changeUserPasswordCli.php [hunk 1]
```diff
@@ -46,16 +46,17 @@ public function __construct($params)
...
-
         $this->_userManager = owa_coreApi::supportClassFactory('base', 'userManager');
-        $rules = $this->_userManager->getPasswordValidationRules($this->getParam('password'));
+    }
+
+    public function validate()
+    {
+        $this->addValidation('user_required', $this->getParam('user'), 'required');
+
+        $validations = $this->_userManager->getPasswordValidationRules($this->getParam('password'));
```
Comment (padams) [inline modules/base/changeUserPasswordCli.php:56]: There's probably no need to keep the validations in userManager any longer :)

File: modules/base/usersChangePassword.php [hunk 2]
```diff
@@ -36,29 +36,16 @@
...
-     * @var owa_userManager
-     */
-    private $_userManager;
+    public function validate()
+    {
+        $this->addValidation('password_match', [$this->getParam('password'), $this->getParam('password2')], 'stringMatch', ['errorMsg' => 'Your passwords must match.']);
 
-    function __construct($params) {
+        $userManager = owa_coreApi::supportClassFactory('base', 'userManager');
+        $validations = $userManager->getPasswordValidationRules($this->getParam('password'));
```
Comment (padams) [inline modules/base/usersChangePassword.php:44]: Maybe we can move the validations back to the controller now.

**Pull request merged**